kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -949,13 +947,22 @@ class ResidualBasedBlockBuilderAndSolver
     virtual void ConstructMatrixStructure(
         typename TSchemeType::Pointer pScheme,
         TSystemMatrixType& A,
-        ElementsContainerType& rElements,
-        ConditionsArrayType& rConditions,
-        ProcessInfo& CurrentProcessInfo)
+        ModelPart& rModelPart)
     {
         //filling with zero the matrix (creating the structure)
         Timer::Start("MatrixStructure");
 
+        // Getting the elements from the model
+        const int nelements = static_cast<int>(rModelPart.Elements().size());
+
+        // Getting the array of the conditions
+        const int nconditions = static_cast<int>(rModelPart.Conditions().size());
+
+        ProcessInfo& CurrentProcessInfo = rModelPart.GetProcessInfo();
+        ModelPart::ElementsContainerType::iterator el_begin = rModelPart.ElementsBegin();
+        ModelPart::ConditionsContainerType::iterator cond_begin = rModelPart.ConditionsBegin();
+
+
         const std::size_t equation_size = BaseType::mEquationSystemSize;
 
 #ifdef USE_GOOGLE_HASH

```

Comment:
A side comment:
@RiccardoRossi with the recent improvements of std::unordered_map performance do we still need this macro section?

Reply:
I think we can remove the GOOGLE_HASH, however it would make sense to eventually merge #1518 for such a fix (and that in turn requires some more testing, particularly to see performance in windows)